crates/ra_hir_expand/src/builtin_macro.rs

```diff
@@ -78,3 +82,26 @@ fn line_expand(
 
     Ok(expanded)
 }
+
+fn stringify_expand(
+    db: &dyn AstDatabase,
+    id: MacroCallId,
+    _tt: &tt::Subtree,
+) -> Result<tt::Subtree, mbe::ExpandError> {
+    let loc = db.lookup_intern_macro(id);
+    let macro_call = loc.ast_id.to_node(db);
+
+    let macro_content = {
+        let arg = macro_call.token_tree().ok_or_else(|| mbe::ExpandError::UnexpectedToken)?;
```

Comment:
Might be a stupid question, but is this the same as `_tt`?

Reply:
There is no spacing information in `_tt` but I agree it is more accurate to use `_tt` directly here :

[Playground Link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=84c7cc6ac3e862543cdbdbea95eb5e48)

The builtin `stringify!` macro do not preserve space too.

Reply:
Might be worth adding a test.

Reply:
I think either solution is fine. It's also a shame that we can't easily add a test here, because there's no testing infra in `hir_expand`, but I think it makes sense to merge this without blocking on the said infra